HIP/hip-850.md

```diff
@@ -0,0 +1,70 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Account
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2024-04-03T07:00:00Z
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks, TierBot
+updated: 2024-03-20
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury account. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury account, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
+
+## Rationale
+
+The enhancement focuses on granting the Supply Key the capability to update the data of an NFT serial number exclusively while it is in the treasury account. This approach ensures security and control, preventing unauthorized modifications once the NFT is distributed.
+
+## User Stories
+
+- As a game developer, I want to update game piece NFTs with player stats during gameplay, ensuring the data becomes immutable once the player regains ownership.
+- As an event organizer, I need to update event ticket NFTs with attendee information while they are in the treasury account, guaranteeing data immutability post-transfer.
+
+## Specification
+
+This HIP requires an update to the Supply Key mechanism, enabling it to modify the metadata of NFTs while they are held in the treasury account. The NFT data should remain immutable once distributed, ensuring integrity and trust in the NFT ecosystem.
+
+## Backwards Compatibility
+
+This proposal introduces no backward incompatibilities within the current Hedera ecosystem.
+
+## Security Implications
```

Comment:
Is there an implication on the mirror node here?
I imagine a modification of metadata needs to be captured so that I can see the history of an NFT metadata.

Reply:
I don't think so. The field will change from mutable from immutable in the update and when the record stream comes through it'll update the metadata record which will still be accessible through pulling that serial number from the mirror node API for that NFTid.

Reply:
Mirror node already (0.101+) tracks NFT metadata updated via the new (HIP-657) `TokenUpdateNftsTransaction`. This is facilitated by signing the transaction with the metadata key defined for the token. This can happen any time, and thus the NFT is not immutable. My understanding is that this can be done regardless of whether or not the NFT is held in the treasury account. More testing will be done as 0.49 and the client SDKs move forward.

Reply:
@ty-swirldslabs can you clarify this.
If HIP 657 allows for metadata modification why is this HIP needed.

If this HIP was implemented the result would be
1. The metadata key can update anytime
2. The supply key can update metadata only when the token serial is in the treasury balance?

Is this correct? If so why is 2 needed when 1 already works?
If this is needed is the real goal creating immutability for the metadata, in this case when the token is in the treasuries balance?

Also would a supply key be able to update the metadata on a token if there was no metadata key (implying immutability?). 

These considerations should be clarified explicitly in this HIP for clarity

Reply:
@Nana-EC 

These HIPs used to be the same HIP, but I was advised to separate them to help with implementation and understanding. They bring two very different functionalities to Hedera Token Service. 

HIP-657:
- Enables remote updating of the metadata field of NFTs in circulation
- All NFTs in collection at all times are update-able when metadata key is present

HIP-850
- Enables treasury updates of NFTs only when held by treasury
- Only NFTs in the treasury wallet are mutable
- All NFTs in the collection in circulation are immutable

657 gets us closer to Smart contract expecations of being able to remotely adaapt metadata at any time, which hedera currently can't do in a decentralized way (centralized server urls would work, but isn't true web3 NFT arch)

850 brings a new paradigm that does not exist anywhere else in Web3 to my knowledge and has vast business use cases. It allows token distributors to gaurentee immutability outside of the treasury, while still being able to update and maintain provenance when the asset is sent back to the distributor. 

Take RWAs being sold on an open market.

If the metadata key was on the token, that RWA could be adapted remotely after it's already been put on a market, or sold to a buyer. This introduces trust that the distributor won't change the assets underlying metadata that dictates it's value. 

However, if the asset didn't have a metadata key, but maintained it's supply key, the distributor would be able to adapt the asset when the owner sends it back to the treasury (house sale, lease update, agreement adaptations of a RWA, etc) but when the owner puts it onto a market, it's guaranteed to be immutable and not adaptable.

I believe this is an extremely powerful dynamic that will help enterprise create functionality for legal needs in the future of tokenization of assets. 

**If HIP 657 allows for metadata modification why is this HIP needed.**

- 657 does not guarantee immutability outside the treasury, leaving need for trust of no remote changes to tokenized asset. 850 provides this unique functionality not see

 **If so why is 2 needed when 1 already works?**
 They provide fundamentally different abilities for token distributors. 657 allows a metadata key to adapt any NFT at any time. 850 only allows NFTs to be adapted in the treasury wallet, guaranteeing immutability when on the market, unlocking functionality that does not  currently exisit on any network I'm aware of. 

**If this is needed is the real goal creating immutability for the metadata, in this case when the token is in the treasuries balance?**
No. It is creating mutability only in the treasury wallet, while guaranteeing immutability outside the treasury wallet. 

**Also would a supply key be able to update the metadata on a token if there was no metadata key (implying immutability?).**
Yes. Metadata key is a completely separate way of adapting data and is not required for the supply key to be able to adapt the metadata of NFts in the treasury wallet.

Reply:
@ty-swirldslabs So as I understand this, this is specifically when the token doesn't have an admin key or metadata key. We will be able to update metadata using supply key _only_ when in treasury account?
Because with HIP-657, we will be able to update metadata of NFT when in treasury account using `metadataKey`

Reply:
That is correct!

Reply:
@ty-swirldslabs I think the HIP needs to explicitly call out which transaction the supply key owner will utilize.
Sounds like its the `TokenUpdateNftsTransaction` . You should note this under the Specification section to note it is reusing an existing HAPI transaction.
It may actually be worth specifying the comments to update that transactions comments to better describe how it can be used